java/src/main/java/com/logicalclocks/hsfs/engine/HudiEngine.java

```diff
@@ -0,0 +1,214 @@
+  private static final String HUDI_QUERY_TYPE_INCREMENTAL_OPT_VAL = "incremental";
+  private static final String HUDI_BEGIN_INSTANTTIME_OPT_KEY = "hoodie.datasource.read.begin.instanttime";
+  private static final String HUDI_END_INSTANTTIME_OPT_KEY  = "hoodie.datasource.read.end.instanttime";
+
+  private static final String PAYLOAD_CLASS_OPT_KEY = "hoodie.datasource.write.payload.class";
+  private static final String PAYLOAD_CLASS_OPT_VAL =  "org.apache.hudi.common.model.EmptyHoodieRecordPayload";
+  private static final String HUDI_DELETE = "delete";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HudiEngine.class);
+  private Utils utils = new Utils();
+  private FeatureGroupApi featureGroupApi = new FeatureGroupApi();
+  private SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMddHHmmss");
+  private FeatureGroupCommit fgCommitMetadata = new FeatureGroupCommit();
+
+
+  public void saveHudiFeatureGroup(SparkSession sparkSession, FeatureGroup featureGroup,
+                                   Dataset<Row> dataset, SaveMode saveMode, HudiOperationType operation,
+                                   Map<String, String> writeOptions)
+      throws IOException, FeatureStoreException {
+
+    Map<String, String> hudiArgs = setupHudiWriteOpts(featureGroup, operation);
+    if (writeOptions != null && !writeOptions.isEmpty()) {
+      hudiArgs.putAll(writeOptions);
+    }
+
+    dataset.write()
+        .format(HUDI_SPARK_FORMAT)
+        .options(hudiArgs)
+        .mode(saveMode)
+        .save(featureGroup.getLocation());
+
+    FeatureGroupCommit fgCommit = getLastCommitMetadata(sparkSession, featureGroup.getLocation());
+
+    featureGroupApi.featureGroupCommit(featureGroup, fgCommit);
+  }
+
+  public FeatureGroupCommit deleteRecord(SparkSession sparkSession, FeatureGroup featureGroup, Dataset<Row> deleteDF)
+      throws IOException, FeatureStoreException {
+
+    Map<String, String> hudiArgs = setupHudiWriteOpts(featureGroup, HudiOperationType.UPSERT);
+
+    deleteDF.write().format(HUDI_SPARK_FORMAT)
+        .options(hudiArgs)
+        .mode(SaveMode.Append)
+        .save(featureGroup.getLocation());
+
+    FeatureGroupCommit fgCommit = getLastCommitMetadata(sparkSession, featureGroup.getLocation());
+    FeatureGroupCommit apiFgCommit = featureGroupApi.featureGroupCommit(featureGroup, fgCommit);
+    apiFgCommit.setCommitID(apiFgCommit.getCommitID());
+
+    return apiFgCommit;
+  }
+
+  public void registerTemporaryTable(SparkSession sparkSession, FeatureGroup featureGroup, String alias,
+                                     Long startTimestamp, Long  endTimestamp) {
+    sparkSession.conf().set("spark.sql.hive.convertMetastoreParquet", "false");
+
+    Map<String, String> hudiArgs = setupHudiReadOpts(startTimestamp, endTimestamp);
+    sparkSession.read()
+        .format(HUDI_SPARK_FORMAT)
+        .options(hudiArgs)
+        .load(featureGroup.getLocation()).registerTempTable(alias);
```

Comment:
`registerTempTable` is deprecated, you should use `createOrReplaceTempView` instead, which takes care of replacing the alias if you run multiple queries in the same Spark session.

Reply:
done